synapseclient/core/utils.py

```diff
@@ -1331,3 +1335,39 @@ def delete_none_keys(incoming_object: typing.Dict) -> None:
         for key in list(incoming_object.keys()):
             if incoming_object[key] is None:
                 del incoming_object[key]
+
+
+def merge_dataclass_entities(
+    source: typing.Union["Project", "Folder"],
```

Comment:
I can see this function being helpful for any Entities that support annotations. (File, Table, Dataset, etc). Not sure if we need to add it in right here, but just a note

Reply:
Do you think that there is an "annotation" mixIn here where the classes that have it can do something like sets in python and have `.update` functionality?

Reply:
Can you elaborate a bit more on your thoughts on this? I see what sets is doing, but I'm unclear what you mean about applying it in this use-case.

Reply:
> `MyFileEntity.update(MySecondFileEntity) and it would just update the annotations.`

I don't think so in this case because we are setting annotations to be a dict. This function was built out of the need that dataclasses didn't have any nifty mechanism I could find that would do a merge.